utils/cache/backend_redis.c

```diff
@@ -227,12 +229,38 @@ static status_t redis_list_pop(redisContext* c, const char* const key, char* res
   return ret;
 }
 
+long int redis_occupied_space(redisContext* c) {
+  const char size_field[] = "used_memory:";
+  redisReply* reply = redisCommand(c, "INFO");
+  if (reply->type == REDIS_REPLY_STRING) {
+    // Parsing the result returned by redis. Get information from field `used_memory`
+    char* str = strstr(reply->str, size_field);
+    char* size_str = strtok(str, "\n");
+    char* ptr = size_str + strlen(size_field);
+
+    char* strtol_p = NULL;
+    long int strtol_temp;
+    strtol_temp = strtol(ptr, &strtol_p, 10);
+    if (strtol_p != ptr && errno != ERANGE && strtol_temp >= INT_MIN && strtol_temp <= INT_MAX) {
+      freeReplyObject(reply);
+      return strtol_temp;
+    } else {
+      ta_log_error("Malformed input or illegal input character\n");
+    }
+  } else {
+    ta_log_error("%s\n", "Failed to operate redis command.");
+  }
+
+  freeReplyObject(reply);
+  return -1;
+}
+
 /*
  * Public functions
  */
 
-bool cache_init(bool state, const char* host, int port) {
-  cache_state = state;
+bool cache_init(bool input_state, const char* host, int port) {
+  state = input_state;
```

Comment:
Can you eliminate the use of global variable `state`?

Reply:
It may cause some issues.
The global variable is used to block any access to redis under the condition that cache service is not turning on, so if we are going to remove this variable, then we still need another way to record the current state of caching service.